### PR TITLE
Release v1.17.0

### PR DESCRIPTION
**Pull request by BeyondEvil**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -28,6 +28,12 @@ basepython = python
 deps = flake8
 commands = flake8 {posargs:.}
 
+[testenv:black]
+skip_install = true
+basepython = python
+deps = black
+commands = black --check {posargs:.}
```
Comment (BeyondEvil) [inline tox.ini]: `--check` only checks the python files for incorrect formatting and will fail the test if unformatted files are found.

**Comment by BeyondEvil** [comment]
I suggest we remove `pre-commit` and put black in tox instead. There's no mention of it in the docs and it requires the contributor to install extra stuff - and we have no way of enforcing it.

Putting it in tox enforces that the formatting is at least compatible with Black.

If we want, we can make black change files when run locally but only do the check when run in Travis.

**Comment by nicoddemus** [comment]
> I suggest we remove pre-commit and put black in tox instead. There's no mention of it in the docs and it requires the contributor to install extra stuff - and we have no way of enforcing it.

FTR on pytest we enforce the pre-commit rules in our tox files:

https://github.com/pytest-dev/pytest/blob/0d3958e8dec5f78ed19ba5bc7a00a35f3ea3deb0/tox.ini#L53-L57

**Comment by RonnyPfannschmidt** [comment]
pre-commit makes the work-flow a breeze for anyone callable of running 2 simple commands,
putting things into other tools with other version pinning semantics is pretty much an absolute and utter pain

**Comment by nicoddemus** [comment]
I concur, and that from a Windows user where using pre-commit is *much* slower than on Linux (not on pre-commmit's fault, but because Windows FS and venvs are much slower than on Linux). 😁

**Comment by BeyondEvil** [comment]
> > I suggest we remove pre-commit and put black in tox instead. There's no mention of it in the docs and it requires the contributor to install extra stuff - and we have no way of enforcing it.
> 
> FTR on pytest we enforce the pre-commit rules in our tox files:
> 
> https://github.com/pytest-dev/pytest/blob/0d3958e8dec5f78ed19ba5bc7a00a35f3ea3deb0/tox.ini#L53-L57

See, I didn't know you could do that. That takes care for the enforcing bit.

Since I don't feel strongly either way, we'll stick with pre-commit. I'll update the docs and and stuffs to tox.

**Comment by BeyondEvil** [comment]
> pre-commit makes the work-flow a breeze for anyone callable of running 2 simple commands,
> putting things into other tools with other version pinning semantics is pretty much an absolute and utter pain

Well, when you put it that way. 😂

**Comment by BeyondEvil** [comment]
@davehunt @nicoddemus @RonnyPfannschmidt 

Please review the changes. Sorry for any extra work this has caused you. 🙏

File: docs/development.rst [hunk 2]
```diff
@@ -1,6 +1,25 @@
 Development
 ===========
 
+To contribute to `pytest-selenium` you can use `Pipenv`_ to manage
+a python virutal environment and `pre-commit <https://pre-commit.com/>`_ to help you with
```
Comment (nicoddemus) [inline docs/development.rst]: * virtual

Comment (BeyondEvil) [reply docs/development.rst]: Thanks, fixed!

**Pull request merged**